whois-api/src/main/java/net/ripe/db/whois/api/rdap/RdapObjectMapper.java

```diff
@@ -495,21 +499,23 @@ private List<Entity> createContactEntities(final RpslObject rpslObject, final St
                 entity.getRoles().add(CONTACT_ATTRIBUTE_TO_ROLE_NAME.get(attributeType));
             }
             mapEntityLinks(entity, requestUrl, entry.getKey());
-            mapEntityVcard(entry, entity, objectPossibleTypes);
+            mapEntityVcardAndContactRedaction(rdapObject, entry, entity, objectPossibleTypes);
             entities.add(entity);
         }
 
-        return entities;
+        rdapObject.getRedacted().addAll(RedactionObjectMapper.createEntityRedaction(rpslObject.getAttributes()));
+        rdapObject.getEntitySearchResults().addAll(entities);
     }
 
-    private void mapEntityVcard(final Map.Entry<CIString, Set<AttributeType>> entry, final Entity entity,
-                                final Set<ObjectType> objectPossibleTypes) {
+    private void mapEntityVcardAndContactRedaction(final RdapObject rdapObject, final Map.Entry<CIString, Set<AttributeType>> entry, final Entity entity,
```

Comment:
rename to mapVcardAndRedaction? Feel free to ignore

Reply:
Done